source/val/Function.h

```diff
@@ -179,6 +192,9 @@ class Function {
   /// The block that is currently being parsed
   BasicBlock* current_block_;
 
+  /// A psudo exit block that is the successor to all return blocks
+  BasicBlock exit_block_;
```

Comment:
Strongly suggest actually calling it pseudo_exit_block_

Also, it's worth saying that by default its ID will be 0.  By SPIR-V rules, that's never a valid ID, so at least the psuedo exit block won't ever compare as equal to any other block's id.